detail/provisioning.cattle.io.cluster.vue

```diff
@@ -55,60 +55,62 @@ export default {
-        hash.nodeTemplates = this.$store.dispatch('management/findAll', { type: MANAGEMENT.NODE_TEMPLATE });
-      }
-    } else {
-      if ( this.$store.getters['management/canList'](CAPI.MACHINE_DEPLOYMENT) ) {
-        hash.machineDeployments = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE_DEPLOYMENT });
-      }
+    if ( this.$store.getters['management/canList'](CAPI.MACHINE_DEPLOYMENT) ) {
+      fetchOne.machineDeployments = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE_DEPLOYMENT });
+    }
 
-      if ( this.$store.getters['management/canList'](CAPI.MACHINE) ) {
-        hash.machines = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE });
-      }
+    if ( this.$store.getters['management/canList'](CAPI.MACHINE) ) {
+      fetchOne.machines = this.$store.dispatch('management/findAll', { type: CAPI.MACHINE });
     }
 
     if (this.value.isImported || this.value.isCustom) {
-      hash.clusterToken = this.value.getOrCreateToken();
+      fetchOne.clusterToken = this.value.getOrCreateToken();
     }
     if ( this.value.isRke1 && this.$store.getters['isRancher'] ) {
-      hash.etcdBackups = this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+      fetchOne.etcdBackups = this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
 
-      hash.normanNodePools = this.$store.dispatch('rancher/findAll', { type: NORMAN.NODE_POOL });
+      fetchOne.normanNodePools = this.$store.dispatch('rancher/findAll', { type: NORMAN.NODE_POOL });
     }
 
-    const res = await allHash(hash);
+    const fetchOneRes = await allHash(fetchOne);
 
-    this.allMachines = res.machines || [];
-    this.allMachineDeployments = res.machineDeployments || [];
-    this.allNodes = res.allNodes || [];
-    this.allNodePools = res.allNodePools || [];
-    this.haveMachines = !!res.machines;
-    this.haveDeployments = !!res.machineDeployments;
-    this.haveNodePools = !!res.allNodePools;
-    this.haveNodes = !!res.allNodes;
+    this.allMachines = fetchOneRes.machines || [];
+    this.allMachineDeployments = fetchOneRes.machineDeployments || [];
+    this.haveMachines = !!fetchOneRes.machines;
+    this.haveDeployments = !!fetchOneRes.machineDeployments;
+    this.clusterToken = fetchOneRes.clusterToken;
+    this.etcdBackups = fetchOneRes.etcdBackups;
 
-    this.clusterToken = res.clusterToken;
-    this.etcdBackups = res.etcdBackups;
+    const fetchTwo = {};
 
-    const machineDeloymentTemplateType = res.machineDeployments?.[0]?.templateType;
+    const machineDeloymentTemplateType = fetchOneRes.machineDeployments?.[0]?.templateType;
 
     if (machineDeloymentTemplateType && this.$store.getters['management/schemaFor'](machineDeloymentTemplateType) ) {
-      await this.$store.dispatch('management/findAll', { type: machineDeloymentTemplateType });
+      fetchTwo.mdtt = this.$store.dispatch('management/findAll', { type: machineDeloymentTemplateType });
```

Comment:
I think we can get away without performing the assignment if `mdtt` won't be used anywhere

edit: but I can see how we might want to await later at line 110.. maybe it makes sense to keep it here

Reply:
Yeah I didn't want to put a another wait for an API request so do it at the same time the nodes lists (may) be fetched.

That's basically the reason the PR is a bit slower. Initially all API calls were executed in parallel and now they're done in two chunks. In theory the change should be minimal as the longer calls will be fetching nodes / machines, think the other won't suffer from scaling issues and should be pretty quick.